### ISSUE TITLE
Windows agent dies after a few days

### ISSUE THREAD

**Issue by feebles303** [sensu/sensu-go#4527]
I've got a few windows server 2019 instances that I'm monitoring with sensu-go and a few days after installing the agent, it dies and needs to be reinstalled to get in going again. It works as expected up until the point it dies.

When I say die, I mean it crashes and then will always crash whether run on the command line or by the windows service manager. Even getting help or the version will crash after it has died.

```
C:\Windows\system32>"c:\Program Files\Sensu\sensu-agent\bin\sensu-agent.exe" version
fatal error: runtime: mcall called on m->g0 stack

runtime stack:
encoding/base64.NewEncoding({0x1eeb43f, 0x40})
        encoding/base64/base64.go:47 +0x36
encoding/base64.init()
        encoding/base64/base64.go:99 +0x29

goroutine 1 [runnable, locked to thread]:
encoding/base64.NewEncoding({0x1eeb43f, 0x40})
        encoding/base64/base64.go:47 +0x36
encoding/base64.init()
        encoding/base64/base64.go:99 +0x29v
```

```
C:\Windows\system32>"c:\Program Files\Sensu\sensu-agent\bin\sensu-agent.exe" help
fatal error: runtime: mcall called on m->g0 stack

runtime stack:
time.init()
        time/format.go:413

goroutine 1 [running, locked to thread]:
time.init()
        time/format.go:413 +0x479 fp=0xc000095aa0 sp=0xc000095a98 pc=0x145d899
runtime.doInit(0x271bf40)
        runtime/proc.go:6498 +0x12a fp=0xc000095bd8 sp=0xc000095aa0 pc=0x140684a
runtime.doInit(0x271afe0)
        runtime/proc.go:6475 +0x71 fp=0xc000095d10 sp=0xc000095bd8 pc=0x1406791
runtime.doInit(0x271a860)
        runtime/proc.go:6475 +0x71 fp=0xc000095e48 sp=0xc000095d10 pc=0x1406791
runtime.doInit(0x271cf60)
        runtime/proc.go:6475 +0x71 fp=0xc000095f80 sp=0xc000095e48 pc=0x1406791
runtime.main()
        runtime/proc.go:238 +0x1d8 fp=0xc000095fe0 sp=0xc000095f80 pc=0x13f9fd8
runtime.goexit()
        runtime/asm_amd64.s:1581 +0x1 fp=0xc000095fe8 sp=0xc000095fe0 pc=0x1425861
```

I installed sensu-go from `sensu-go-agent_6.6.0.5502_en-US.x64.msi` and that's the version the above crashes are from but had the same problem with 6.5.5, 6.5.4 and 6.5.1

`sensu-backend version` gives
`sensu-backend version 6.5.1+ee, enterprise edition, build 498468e022d998bad8a1f922a0833ecf2e3c8bdf, built 2021-10-20T20:23:36Z, built with go1.17.1`

**Comment by amdprophet** [comment]
Hi @feebles303,

Has this behaviour happened on any versions before 6.5.x? Can you tell me more about your instances (CPU, Cloud, AMI ID, etc. if available)? I've never seen this before before and I'd like to try to replicate the environment you're using.

**Comment by feebles303** [comment]
Hi @amdprophet,

No I hadn't seen this until I upgraded everything to 6.5, I can't remember the exact version that I had before that but I think it was 6.1 or 6.2.

The instances are VMs running on a small Joyent/Triton on-premise cloud (see https://www.joyent.com/triton/on-premise). Essentially the hypervisor is bhyve and the host OS is SmartOS. The CPU on the host of the VM that's currently failing is AMD EPYC 7542 32-Core. The VM is allocated 16 cores.

**Comment by feebles303** [comment]
Yes, those stack dumps do look like they could be from go internals.

Yes, I can test 6.4.x, I've installed 6.4.2 on the VM that the sensu-agent was crashing on. Will give that a week or so and see how it goes.

If that resolves the issue then I'm happy to stick with 6.4.2 until you upgrade go again, then I can give that a test.

**Comment by asachs01** [comment]
@feebles303 it's been some time since you originally opened this issue and we're going to close out this issue. If it's still occurring, feel free to reopen/comment.
